File: src/features/typing/data/go/n_queens.go
package main

func solveNQueens(n int) [][]string {
	result := [][]string{}
	queens := make([]int, n)
	cols := make(map[int]bool)
	diag1 := make(map[int]bool)
	diag2 := make(map[int]bool)
	var backtrack func(row int)
	backtrack = func(row int) {
		if row == n {
			board := make([]string, n)
			for i := 0; i < n; i++ {
				line := make([]byte, n)
				for j := 0; j < n; j++ {
					line[j] = '.'
				}
				line[queens[i]] = 'Q'
				board[i] = string(line)
			}
			result = append(result, board)
			return
		}
		for col := 0; col < n; col++ {
			if cols[col] || diag1[row+col] || diag2[row-col] {
				continue
			}
			queens[row] = col
			cols[col] = true
			diag1[row+col] = true
			diag2[row-col] = true
			backtrack(row + 1)
			cols[col] = false
			diag1[row+col] = false
			diag2[row-col] = false
		}
	}
	backtrack(0)
	return result
}

File: src/features/typing/data/dart/n_queens.dart
class Solution {
  List<List<String>> solveNQueens(int n) {
    final result = <List<String>>[];
    final queens = List<int>.filled(n, 0);
    final cols = <int>{};
    final diag1 = <int>{};
    final diag2 = <int>{};
    void backtrack(int row) {
      if (row == n) {
        final board = <String>[];
        for (int i = 0; i < n; i++) {
          final line = List<String>.filled(n, '.');
          line[queens[i]] = 'Q';
          board.add(line.join());
        }
        result.add(board);
        return;
      }
      for (int col = 0; col < n; col++) {
        if (cols.contains(col) ||
            diag1.contains(row + col) ||
            diag2.contains(row - col)) {
          continue;
        }
        queens[row] = col;
        cols.add(col);
        diag1.add(row + col);
        diag2.add(row - col);
        backtrack(row + 1);
        cols.remove(col);
        diag1.remove(row + col);
        diag2.remove(row - col);
      }
    }
    backtrack(0);
    return result;
  }
}
